addons/website_quote/models/order.py

```diff
@@ -129,6 +138,18 @@ def write(self, cr, uid, ids, values, context=None):
         values = self._inject_quote_description(cr, uid, values, context)
         return super(sale_order_line, self).write(cr, uid, ids, values, context=context)
 
+    # The overrided onchange was written in new api, sorry
+    # Take the description on the order template if the product is present in it
+    @api.multi
+    @api.onchange('product_id')
+    def product_id_change(self):
+        domain = super(sale_order_line, self).product_id_change()
+        if self.order_id.template_id:
+            for quote_line in self.order_id.template_id.quote_line:
```

Comment:
self.name = next(quote_line.name for quote_line in self.order_id.template_id.quote_line if quote_line.product_id == self.product_id) could work. Maybe. Just to show you can use next.